spec/lib/next_gen_images_spec.rb

```diff
@@ -5,64 +5,75 @@
+        it { is_expected.to include("<picture>#{source(original_src_url, 'image/jpeg')}<img src=\"#{original_src_url}\"></picture>") }
       end
 
       context "when the image is already in a picture tag" do
-        let(:body) { "<picture><img src=\"#{original_src}\"></picture>" }
+        let(:body) { "<picture><img src=\"#{original_src_url}\"></picture>" }
 
         it { is_expected.to include(body) }
       end
 
       context "when there are next-gen images" do
         let(:next_gen_imgs) do
           {
-            "image/webp" => original_src.gsub(original_ext, ".webp"),
-            "image/jp2" => original_src.gsub(original_ext, ".jp2"),
-            "image/jpeg" => original_src,
+            "image/webp" => original_src_path.gsub(original_ext, ".webp"),
+            "image/jp2" => original_src_path.gsub(original_ext, ".jp2"),
+            "image/jpeg" => original_src_path,
           }
         end
 
         before do
           next_gen_imgs.values.each do |src|
-            allow(File).to receive(:exist?).with("#{Rails.public_path}/#{src}").and_return(true)
+            allow(File).to receive(:exist?).with("#{Rails.public_path}#{src}").and_return(true)
           end
         end
 
         it do
           sources = next_gen_imgs.map { |mime_type, src| source(src, mime_type) }.join
-          is_expected.to include("<picture>#{sources}<img src=\"#{original_src}\"></picture>")
+          is_expected.to include("<picture>#{sources}<img src=\"#{original_src_url}\"></picture>")
+        end
+
+        context "when the src is absolute (assets are hosted on a different domain)" do
+          let(:asset_host) { "https://domain.com" }
+
+          it do
+            sources = next_gen_imgs.map { |mime_type, src| source(src, mime_type) }.join
+            is_expected.to include("<picture>#{sources}<img src=\"#{original_src_url}\"></picture>")
+          end
         end
       end
     end
 
     context "when the original image is a png" do
-      let(:original_src) { "path/to/image.png" }
+      let(:original_src_path) { "/path/to/image.png" }
 
-      it { is_expected.to include("<picture>#{source(original_src, 'image/png')}<img src=\"#{original_src}\"></picture>") }
+      it { is_expected.to include("<picture>#{source(original_src_path, 'image/png')}<img src=\"#{original_src_url}\"></picture>") }
     end
 
     context "when the original image is an svg" do
-      let(:original_src) { "path/to/image.svg" }
+      let(:original_src_path) { "/path/to/image.svg" }
 
-      it { is_expected.to include("<picture>#{source(original_src, 'image/svg+xml')}<img src=\"#{original_src}\"></picture>") }
+      it { is_expected.to include("<picture>#{source(original_src_path, 'image/svg+xml')}<img src=\"#{original_src_url}\"></picture>") }
     end
   end
 
   def source(src, mime_type)
-    "<source srcset=\"#{src}\" type=\"#{mime_type}\"></source>"
+    "<source srcset=\"#{asset_host}#{src}\" type=\"#{mime_type}\"></source>"
```

Comment:
worth using an alternative delimiter to avoid escaping?

```suggestion
    %(<source srcset="#{asset_host}#{src}" type="#{mime_type}"></source>)
```

Reply:
I'll never get used to that as a string delimiter, but yeah agreed I'll push an update 😅